query-grammar/src/infallible.rs

```diff
@@ -0,0 +1,286 @@
+//! nom combinators for infallible operations
+
+use nom::{IResult, InputLength};
+pub type ErrorList = Vec<(usize, String)>;
```

Comment:
No clue what the first `usize` argument is.

Reply:
Can we introduce type to make it less confusing?

It seems like the first argument changes meaning halfway through. (offset to the end to offset from the start)